src/Form/Form.js

```diff
@@ -174,22 +202,35 @@ class Form {
 
     get submitButtons () {
         const selector = this.matching.cssSelector('SUBMIT_BUTTON_SELECTOR')
-        return [...this.form.querySelectorAll(selector)]
-            .filter((button) => {
-                const content = button.textContent || ''
-                const ariaLabel = button.getAttribute('aria-label') || ''
-                // @ts-ignore
-                const title = button.title || ''
-                // trying to exclude the little buttons to show and hide passwords
-                return !/password|show|toggle|reveal|hide/i.test(content + ariaLabel + title)
-            })
+        const allButtons = /** @type {HTMLElement[]} */([...this.form.querySelectorAll(selector)])
+
+        const likelySubmitButton = allButtons.find(isLikelyASubmitButton)
+        if (likelySubmitButton) return [likelySubmitButton]
```

Comment:
We are now checking for likely submit buttons first to minimise risk of false positive submissions.

Reply:
**nit:**

@GioSensation let's not block on it right now of course, but we might be able to avoid the 2 separate ways of checking here - not for perf or anything like that, but just to clarify intent.